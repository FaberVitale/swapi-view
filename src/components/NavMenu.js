//@flow
import * as React from "react";
import { endpoints, firstPageOf, paths } from "../routes";
import NavigationLink from "./NavigationLink";
import Divider from "material-ui/Divider";

type Props = {
  visible: boolean,
  url: string,
  onLinkClick: () => void
};

const linkSelector = 'a:not([aria-disabled="true"])';
const clickableSelector = `${linkSelector}, ${linkSelector} *`;

class NavMenu extends React.Component<Props> {
  // we need to produce the same markup on the first render because it
  // has to match the shell in index.html
  static isLinkDisabled = (url: string, href: string) => {
    return url === href;
  };

  searchLink: string = endpoints.length > 0 ? `/${endpoints[0]}/search` : "";

  handleClick = ({ button, target, currentTarget }: SyntheticMouseEvent<*>) => {
    if (
      button ||
      target === currentTarget ||
      !(target instanceof HTMLElement) ||
      !target.matches(clickableSelector)
    ) {
      return;
    }
    this.props.onLinkClick();
  };

  render() {
    const { visible, url } = this.props;

    const homeProps = {
      key: 0,
      label: "Home",
      to: paths.home,
      disabled: NavMenu.isLinkDisabled(url, paths.home),
      tabbable: visible,
      fullWidth: true
    };

    const searchProps = {
      key: 1,
      label: "Search",
      to: this.searchLink,
      disabled: NavMenu.isLinkDisabled(url, this.searchLink),
      tabbable: visible,
      fullWidth: true
    };

    const aboutProps = {
      to: paths.about,
      disabled: paths.about === url,
      tabbable: visible,
      fullWidth: true,
      label: "About"
    };

    const list = [
      <NavigationLink {...homeProps} />,
      <NavigationLink {...searchProps} />
    ];

    for (let i = 0, len = endpoints.length; i < len; i++) {
      let endpoint = endpoints[i];
      const disabled = NavMenu.isLinkDisabled(url, firstPageOf[endpoint]);

      list.push(
        <NavigationLink
          key={endpoint}
          to={firstPageOf[endpoint]}
          disabled={disabled}
          label={endpoint}
          fullWidth={true}
          tabbable={visible}
        />
      );
    }

    return (
      <nav
        className="sidebar-nav flex-column justify-space-between"
        onClick={this.handleClick}
      >
        <div className="no-pointer-events">{list}</div>
        <div className="sidebar-bottom no-pointer-events">
          <Divider />
          <NavigationLink {...aboutProps} />
        </div>
      </nav>
    );
  }
}

export default NavMenu;
