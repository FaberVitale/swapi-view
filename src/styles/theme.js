//@flow
import {
  yellow200 as primary1,
  yellow500 as primary2,
  grey400 as accent3
} from "material-ui/styles/colors";
import darkBaseTheme from "material-ui/styles/baseThemes/darkBaseTheme";
import getMuiTheme from "material-ui/styles/getMuiTheme";

const boxShadow =
  "0 1px 6px rgba(0, 0, 0, 0.12), 0 1px 4px rgba(0, 0, 0, 0.12)";

const theme = getMuiTheme(darkBaseTheme, {
  palette: {
    primary1Color: primary1,
    primary2Color: primary2,
    accent3Color: accent3,
    accent1Color: primary1,
    alternateTextColor: darkBaseTheme.palette.canvasColor
  },
  topBar: {
    backgroundColor: darkBaseTheme.palette.canvasColor,
    padding: 0,
    margin: 0,
    boxShadow: boxShadow
  },

  paper: {
    boxShadow
  },

  homePaper: {
    backgroundColor: "#000000"
  },

  extLink: {
    color: primary1,
    outlineColor: primary2
  },

  h1: {
    color: primary2,
    fontSize: 24
  },
  h2: {
    color: primary2
  },

  noResults: {
    color: "rgba(255, 255, 255, .5)"
  },

  autocomplete: {
    icon: {
      fill: primary1
    }
  },

  textField: {
    floatingLabelColor: "#fefefe",
    focusColor: primary2,
    borderColor: "#fefefe",
    hintColor: "rgba(255, 255, 255, .5)"
  },

  dropDownMenu: {
    accentColor: "#fefefe",
    focusColor: primary2
  },

  h3: darkBaseTheme.h4,

  p: {
    color: primary2
  },
  failedToLoad: {
    frown: {
      color: primary2,
      opacity: 0.1
    }
  },
  menuItem: {
    selectedTextColor: primary2
  }
});

export default theme;
