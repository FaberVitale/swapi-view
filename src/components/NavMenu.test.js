import React from "react";
import NavMenu from "./NavMenu";
import Renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import theme from "../styles/theme";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";

describe("components/NavMenu", () => {
  const tree = Renderer.create(
    <MuiThemeProvider muiTheme={theme}>
      <MemoryRouter initialEntries={["/"]} initialIndex={0}>
        <NavMenu url="/" visible={true} onLinkClick={() => {}} />
      </MemoryRouter>
    </MuiThemeProvider>
  ).toJSON();

  it("matches the snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
});
