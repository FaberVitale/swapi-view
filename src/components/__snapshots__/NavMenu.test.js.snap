// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NavMenu matches the snapshot 1`] = `
<nav
  className="sidebar-nav flex-column justify-space-between"
  onClick={[Function]}
>
  <div
    className="no-pointer-events"
  >
    <a
      aria-disabled={true}
      className="no-pointer-events"
      disabled={true}
      href="/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 0.3)",
          "cursor": "default",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "100%",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={-1}
    >
      <span
        style={
          Object {
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0,
            "muiPrepared": true,
            "paddingLeft": 16,
            "paddingRight": 16,
            "position": "relative",
            "textTransform": "uppercase",
            "verticalAlign": "middle",
          }
        }
      >
        Home
      </span>
    </a>
    <a
      className="pointer-events-auto"
      disabled={false}
      href="/people/search"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 1)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "100%",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          Search
        </span>
      </div>
    </a>
    <a
      className="pointer-events-auto"
      disabled={false}
      href="/people/page/1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 1)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "100%",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          people
        </span>
      </div>
    </a>
    <a
      className="pointer-events-auto"
      disabled={false}
      href="/planets/page/1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 1)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "100%",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          planets
        </span>
      </div>
    </a>
    <a
      className="pointer-events-auto"
      disabled={false}
      href="/films/page/1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 1)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "100%",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          films
        </span>
      </div>
    </a>
    <a
      className="pointer-events-auto"
      disabled={false}
      href="/species/page/1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 1)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "100%",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          species
        </span>
      </div>
    </a>
    <a
      className="pointer-events-auto"
      disabled={false}
      href="/vehicles/page/1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 1)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "100%",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          vehicles
        </span>
      </div>
    </a>
    <a
      className="pointer-events-auto"
      disabled={false}
      href="/starships/page/1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 1)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "100%",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          starships
        </span>
      </div>
    </a>
  </div>
  <div
    className="sidebar-bottom no-pointer-events"
  >
    <hr
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
    <a
      className="pointer-events-auto"
      disabled={false}
      href="/about"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(255, 255, 255, 1)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": "100%",
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          About
        </span>
      </div>
    </a>
  </div>
</nav>
`;
